airflow/jobs/scheduler_job.py

```diff
@@ -1239,6 +1239,8 @@ def _process_executor_events(self, session: Session = None) -> int:
                     "task says its %s. (Info: %s) Was the task killed externally?"
                 )
                 self.log.error(msg, ti, state, ti.state, info)
+                self.log.info('Setting task instance %s state to %s as reported by executor', ti, state)
+                ti.set_state(state)
```

Comment:
I think if you moved these two lines to below the `request =` then the msg would make more sense.

Reply:
Yes. Done. Thanks!